todo-mvc/src/actions/todoStoreActions.ts

```diff
@@ -0,0 +1,47 @@
+import createAction from 'dojo-actions/createAction';
+import Promise from 'dojo-shim/Promise';
+
+import todoStore, { Item } from '../stores/todoStore';
+
+export const addTodo = createAction({
+	do({ label }: { label: string }) {
+		return todoStore.add({ id: `${Date.now()}`, label });
+	}
+});
+
+export const deleteTodo = createAction({
+	do({ id }: { id: string }) {
+		return todoStore.delete(id);
+	}
+});
+
+export const deleteCompleted = createAction({
+	do() {
+		return todoStore.get()
+			// <any> hammer since the store isn't typed to return an iterator (which it does).
+			.then((items: any) => {
+				const promises = Array.from<Item>(items)
+					.filter(({ completed }) => completed)
+					.map(({ id }) => todoStore.delete(id));
+				return Promise.all(promises);
+			});
+	}
+});
+
+export const toggleAll = createAction({
+	do({ checked: completed }: { checked: boolean }) {
+		return todoStore.get()
+			// <any> hammer since the store isn't typed to return an iterator (which it does).
```

Comment:
Actually it seems to return an `IteratorShim` and I still need the hammer. @kitsonk ?